src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/ShadowKmpPlugin.kt

```diff
@@ -18,6 +23,22 @@ public abstract class ShadowKmpPlugin : Plugin<Project> {
             listOf(configurations.getByName(kotlinJvmMain.get().runtimeDependencyConfigurationName))
           },
         )
+        configureMainClass(task)
+      }
+    }
+  }
+
+  private fun configureMainClass(task: ShadowJar) {
+    if (KotlinVersion.CURRENT < KotlinVersion(2, 10, 20)) return
+
+    @OptIn(ExperimentalKotlinGradlePluginApi::class)
+    kmpExtension.jvm().mainRun {
+      task.inputs.property("mainClassName", mainClass)
```

Comment:
Strange, this could be accessed from Kotlin DSL but Groovy...

https://scans.gradle.com/s/7tnldxluomzrc/tests/overview?outcome=FAILED

Reply:
1478b5fb19b7b2f2095c4f4b9d869d010769574c